lib/change_detection/dirty_checking_change_detector.dart

```diff
@@ -462,14 +463,14 @@ class DirtyCheckingRecord<H> implements WatchRecord<H> {
         _getter = null,
         _mode = _MODE_MARKER_;
 
-  dynamic get object => _object;
+  Object get object => _object;
 
   /**
    * Setting an [object] will cause the setter to introspect it and place
    * [DirtyCheckingRecord] into different access modes. If Object it sets up
    * reflection. If [Map] then it sets up map accessor.
    */
-  void set object(Object object) {
+  set object(object) {
```

Comment:
This makes the signature consistent with the base class. If you'd prefer the base class to take
Object object instead of dynamic object let me know and I can change the implementations in that direction. Doesn't really matter much as Object and dynamic are generally interchangeable.